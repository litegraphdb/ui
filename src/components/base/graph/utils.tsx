import { NodeData } from '@/lib/graph/types';
import { defaultNodeColor, nodeLightColorByType, universalNodeColors } from './constant';

const maxLegends = 16;

export const getLegendsForNodes = (nodes: NodeData[]) => {
  const priorityLegends = new Map<string, { legend: string; color: string }>();
  const customLegends = new Map<string, { legend: string; color: string }>();

  // Separate legends into priority (nodeLightColorByType) and custom
  nodes.forEach((node) => {
    if (Object.keys(nodeLightColorByType).includes(node.type)) {
      priorityLegends.set(node.type, { legend: node.type, color: nodeLightColorByType[node.type] });
    } else {
      // Only add if not already present to avoid changing colors for same legend
      if (!customLegends.has(node.type)) {
        customLegends.set(node.type, {
          legend: node.type,
          color: universalNodeColors[customLegends.size % universalNodeColors.length],
        });
      }
    }
  });

  const priorityLegendsSize = priorityLegends.size;
  // If there are nodes with unknown types, add "Unknown" to priority legends
  const isMoreThanMaxLegends = customLegends.size + priorityLegendsSize > maxLegends;

  // Convert to arrays for easier manipulation
  const priorityArray = Array.from(priorityLegends.values());
  const customArray = Array.from(customLegends.values()).slice(
    0,
    maxLegends - (priorityArray.length + (isMoreThanMaxLegends ? 1 : 0))
  );

  // Combine priority legends first, then custom legends
  const legends = [
    ...priorityArray,
    ...customArray,
    ...(isMoreThanMaxLegends ? [{ legend: 'Unknown', color: defaultNodeColor }] : []),
  ];
  const mapLegends: Record<string, { legend: string; color: string }> = {};
  legends.forEach((legend) => {
    mapLegends[legend.legend] = legend;
  });
  return mapLegends;
};
