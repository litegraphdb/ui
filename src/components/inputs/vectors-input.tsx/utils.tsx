export const convertVectorsToAPIRecord = (
  vectors?: Array<{ Model: string; Dimensionality: number; Content: string; Vectors: number[] }>
) => {
  return (
    vectors?.map((vector: any) => ({
      ...vector,
      Dimensionality: Number(vector.Dimensionality),
      Vectors: vector.Vectors || [],
    })) || []
  );
};
