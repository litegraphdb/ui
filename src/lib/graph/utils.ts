import type { NodeData, Point } from './types';

export function isPointInNode(point: Point, node: NodeData, radius: number = 10): boolean {
  const dx = point.x - node.x;
  const dy = point.y - node.y;
  return dx * dx + dy * dy <= radius * radius;
}

export function getNodeAtPosition(nodes: NodeData[], point: Point): NodeData | null {
  return nodes.find((node) => isPointInNode(point, node)) || null;
}

export function transformToOptions<T extends { GUID: string; name?: string; Name?: string }>(
  data?: T[] | null,
  labelField: keyof T = 'name' // Field to use for label in options
) {
  return data
    ? data?.map((item) => ({
        value: item.GUID,
        label: (item[labelField] as string) || item.Name || item.GUID,
      }))
    : [];
}
