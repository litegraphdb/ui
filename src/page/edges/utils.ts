import { EdgeType } from '@/lib/store/edge/types';
import { NodeType } from '@/lib/store/node/types';

// Utility to find the name by GUID
export const getNodeNameByGUID = (guid: string, nodesList: NodeType[]) => {
  if (!guid) return 'N/A';
  const node = nodesList.find((n: NodeType) => n.GUID === guid);
  return node ? node.Name : guid;
};

export const transformEdgeDataForTable = (edgesList: EdgeType[], nodesList: NodeType[]) => {
  return (
    edgesList?.map((data: EdgeType) => {
      return {
        ...data,
        FromName: getNodeNameByGUID(data.From, nodesList),
        ToName: getNodeNameByGUID(data.To, nodesList),
        key: data.GUID,
      } as EdgeType;
    }) || []
  );
};
