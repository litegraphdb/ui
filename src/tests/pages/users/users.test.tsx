import '@testing-library/jest-dom';
import React from 'react';
import { screen, fireEvent, waitFor } from '@testing-library/react';
import UserPage from '@/app/admin/dashboard/users/page';
import { createMockInitialState } from '../../store/mockStore';
import { mockTenantGUID, mockUserData } from '../mockData';
import { handlers } from './handler';
import { setupServer } from 'msw/node';
import { setTenant } from '@/lib/sdk/litegraph.service';
import { commonHandlers } from '@/tests/handler';
import { renderWithRedux } from '@/tests/store/utils';
import AddEditUser from '@/page/users/components/AddEditUser';

const server = setupServer(...handlers, ...commonHandlers);
setTenant(mockTenantGUID);

describe('UsersPage', () => {
  beforeAll(() => server.listen());
  afterEach(() => {
    server.resetHandlers();
  });
  afterAll(() => server.close());

  it('renders the users page', async () => {
    const initialState = createMockInitialState();
    const { container } = renderWithRedux(<UserPage />, initialState, undefined, true);

    await waitFor(() => {
      expect(screen.getAllByText(mockUserData[0].FirstName).length).toBe(2);
    });
    expect(container).toMatchSnapshot('initial table state');
  });

  it('should display Create User button', () => {
    const initialState = createMockInitialState();
    const { container } = renderWithRedux(<UserPage />, initialState, undefined, true);

    const createButton = screen.getByRole('button', { name: /create user/i });
    expect(createButton).toBeVisible();
    expect(createButton).toMatchSnapshot();
  });

  it('should create a user and should be visible in the table', async () => {
    const initialState = createMockInitialState();
    const { container } = renderWithRedux(<UserPage />, initialState, undefined, true);

    await waitFor(() => {
      expect(screen.getAllByText(mockUserData[0].FirstName).length).toBe(2);
    });

    const createButton = screen.getByRole('button', { name: /create user/i });
    expect(createButton).toMatchSnapshot('create user button');
    fireEvent.click(createButton);

    const createModal = screen.getByRole('dialog');
    expect(createModal).toMatchSnapshot('create user modal');

    const nameInput = screen.getByPlaceholderText(/enter first name/i);
    const lastNameInput = screen.getByPlaceholderText(/enter last name/i);
    const emailInput = screen.getByPlaceholderText(/email/i);
    const passwordInput = screen.getByPlaceholderText(/enter password/i);
    const activeInput = screen.getByTestId('active-switch');

    fireEvent.change(nameInput, { target: { value: 'Test User' } });
    fireEvent.change(lastNameInput, { target: { value: 'Test Last Name' } });
    fireEvent.change(emailInput, { target: { value: 'test@example.com' } });
    fireEvent.change(passwordInput, { target: { value: 'testpassword' } });
    fireEvent.click(activeInput);

    const submitButton = screen.getByRole('button', { name: /ok/i });
    fireEvent.click(submitButton);

    const finalTable = container.querySelector('.ant-table');
    expect(finalTable).toMatchSnapshot('final table state after creation');
  });

  it('should update user successfully', async () => {
    const initialState = createMockInitialState();
    const { container } = renderWithRedux(<AddEditUser isAddEditUserVisible={true} setIsAddEditUserVisible={() => {}} user={mockUserData[0]} />, initialState, undefined, true);

    const nameInput = screen.getByPlaceholderText(/enter first name/i);
    const lastNameInput = screen.getByPlaceholderText(/enter last name/i);
    const emailInput = screen.getByPlaceholderText(/email/i);
    const passwordInput = screen.getByPlaceholderText(/enter password/i);
    const activeInput = screen.getByTestId('active-switch');

    fireEvent.change(nameInput, { target: { value: 'Updated User Name' } });
    fireEvent.change(lastNameInput, { target: { value: 'Updated Last Name' } });
    fireEvent.change(emailInput, { target: { value: 'updated@example.com' } });
    fireEvent.change(passwordInput, { target: { value: 'updatedpassword' } });
    fireEvent.click(activeInput);

    const submitButton = screen.getByRole('button', { name: /ok/i });
    fireEvent.click(submitButton);

    const finalTable = container.querySelector('.ant-table');
    expect(finalTable).toMatchSnapshot('final table state after update');
  });

  it('should delete user successfully', async () => {
    const initialState = createMockInitialState();
    const { container } = renderWithRedux(<UserPage />, initialState, undefined, true);

    await waitFor(() => {
      expect(screen.getAllByText(mockUserData[0].FirstName).length).toBe(2);
    });

    const menuButtons = screen.getAllByRole('user-action-menu');
    expect(menuButtons[0]).toBeInTheDocument();
    fireEvent.click(menuButtons[0]);

    const deleteOption = screen.getByText('Delete');
    expect(deleteOption).toBeInTheDocument();
    fireEvent.click(deleteOption);

    const confirmModal = screen.getByRole('dialog');
    expect(confirmModal).toMatchSnapshot('delete confirmation modal');

    const confirmButton = screen.getByRole('button', { name: /delete/i });
    fireEvent.click(confirmButton);

    const finalTable = container.querySelector('.ant-table');
    expect(finalTable).toMatchSnapshot('final table state after deletion');
  });
});