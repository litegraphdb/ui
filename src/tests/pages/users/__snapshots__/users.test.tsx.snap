// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UsersPage renders the users page: initial table state 1`] = `
<div>
  <main
    class="ant-layout-content pageContainer css-dev-only-do-not-override-98ntnt"
    id="users"
  >
    <div
      class="pageTitleContainer ant-flex css-dev-only-do-not-override-98ntnt ant-flex-wrap-wrap ant-flex-align-center ant-flex-justify-space-between ant-flex-gap-small"
    >
      <span
        class="ant-typography css-dev-only-do-not-override-98ntnt"
        data-testid="heading"
        style="font-weight: 600; font-size: 16px;"
      >
        Users
      </span>
      <button
        class="ant-btn css-dev-only-do-not-override-98ntnt ant-btn-link"
        style="font-weight: 500;"
        type="button"
      >
        <span
          class="ant-btn-icon"
        >
          <span
            aria-label="plus-square"
            class="anticon anticon-plus-square"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="plus-square"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M328 544h152v152c0 4.4 3.6 8 8 8h48c4.4 0 8-3.6 8-8V544h152c4.4 0 8-3.6 8-8v-48c0-4.4-3.6-8-8-8H544V328c0-4.4-3.6-8-8-8h-48c-4.4 0-8 3.6-8 8v152H328c-4.4 0-8 3.6-8 8v48c0 4.4 3.6 8 8 8z"
              />
              <path
                d="M880 112H144c-17.7 0-32 14.3-32 32v736c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V144c0-17.7-14.3-32-32-32zm-40 728H184V184h656v656z"
              />
            </svg>
          </span>
        </span>
        <span>
          Create User
        </span>
      </button>
    </div>
    <div
      class="pageContent"
    >
      <div
        class="ant-table-wrapper css-dev-only-do-not-override-98ntnt"
      >
        <div
          class="ant-spin-nested-loading css-dev-only-do-not-override-98ntnt"
        >
          <div
            class="ant-spin-container"
          >
            <div
              class="ant-table css-dev-only-do-not-override-98ntnt"
            >
              <div
                class="ant-table-container"
              >
                <div
                  class="ant-table-content"
                >
                  <table
                    style="table-layout: auto;"
                  >
                    <colgroup>
                      <col
                        style="width: 350px;"
                      />
                      <col
                        style="width: 200px;"
                      />
                      <col
                        style="width: 200px;"
                      />
                      <col
                        style="width: 200px;"
                      />
                      <col
                        style="width: 200px;"
                      />
                      <col
                        style="width: 100px;"
                      />
                      <col
                        style="width: 200px;"
                      />
                      <col
                        style="width: 100px;"
                      />
                    </colgroup>
                    <thead
                      class="ant-table-thead"
                    >
                      <tr>
                        <th
                          class="ant-table-cell react-resizable"
                          scope="col"
                        >
                          <div
                            class="ant-table-filter-column"
                          >
                            <span
                              class="ant-table-column-title"
                            >
                              GUID
                            </span>
                            <span
                              class="ant-dropdown-trigger ant-table-filter-trigger"
                              role="button"
                              tabindex="-1"
                            >
                              <span
                                aria-label="filter"
                                class="anticon anticon-filter"
                                role="img"
                              >
                                <svg
                                  aria-hidden="true"
                                  data-icon="filter"
                                  fill="currentColor"
                                  focusable="false"
                                  height="1em"
                                  viewBox="64 64 896 896"
                                  width="1em"
                                >
                                  <path
                                    d="M349 838c0 17.7 14.2 32 31.8 32h262.4c17.6 0 31.8-14.3 31.8-32V642H349v196zm531.1-684H143.9c-24.5 0-39.8 26.7-27.5 48l221.3 376h348.8l221.3-376c12.1-21.3-3.2-48-27.7-48z"
                                  />
                                </svg>
                              </span>
                            </span>
                          </div>
                          <span
                            class="react-resizable-handle"
                          />
                        </th>
                        <th
                          aria-label="First Name"
                          class="ant-table-cell ant-table-column-has-sorters react-resizable"
                          scope="col"
                          tabindex="0"
                        >
                          <div
                            class="ant-table-filter-column"
                          >
                            <span
                              class="ant-table-column-title"
                            >
                              <div
                                class="ant-table-column-sorters"
                              >
                                <span
                                  class="ant-table-column-title"
                                >
                                  First Name
                                </span>
                                <span
                                  class="ant-table-column-sorter ant-table-column-sorter-full"
                                >
                                  <span
                                    aria-hidden="true"
                                    class="ant-table-column-sorter-inner"
                                  >
                                    <span
                                      aria-label="caret-up"
                                      class="anticon anticon-caret-up ant-table-column-sorter-up"
                                      role="img"
                                    >
                                      <svg
                                        aria-hidden="true"
                                        data-icon="caret-up"
                                        fill="currentColor"
                                        focusable="false"
                                        height="1em"
                                        viewBox="0 0 1024 1024"
                                        width="1em"
                                      >
                                        <path
                                          d="M858.9 689L530.5 308.2c-9.4-10.9-27.5-10.9-37 0L165.1 689c-12.2 14.2-1.2 35 18.5 35h656.8c19.7 0 30.7-20.8 18.5-35z"
                                        />
                                      </svg>
                                    </span>
                                    <span
                                      aria-label="caret-down"
                                      class="anticon anticon-caret-down ant-table-column-sorter-down"
                                      role="img"
                                    >
                                      <svg
                                        aria-hidden="true"
                                        data-icon="caret-down"
                                        fill="currentColor"
                                        focusable="false"
                                        height="1em"
                                        viewBox="0 0 1024 1024"
                                        width="1em"
                                      >
                                        <path
                                          d="M840.4 300H183.6c-19.7 0-30.7 20.8-18.5 35l328.4 380.8c9.4 10.9 27.5 10.9 37 0L858.9 335c12.2-14.2 1.2-35-18.5-35z"
                                        />
                                      </svg>
                                    </span>
                                  </span>
                                </span>
                              </div>
                            </span>
                            <span
                              class="ant-dropdown-trigger ant-table-filter-trigger"
                              role="button"
                              tabindex="-1"
                            >
                              <span
                                aria-label="filter"
                                class="anticon anticon-filter"
                                role="img"
                              >
                                <svg
                                  aria-hidden="true"
                                  data-icon="filter"
                                  fill="currentColor"
                                  focusable="false"
                                  height="1em"
                                  viewBox="64 64 896 896"
                                  width="1em"
                                >
                                  <path
                                    d="M349 838c0 17.7 14.2 32 31.8 32h262.4c17.6 0 31.8-14.3 31.8-32V642H349v196zm531.1-684H143.9c-24.5 0-39.8 26.7-27.5 48l221.3 376h348.8l221.3-376c12.1-21.3-3.2-48-27.7-48z"
                                  />
                                </svg>
                              </span>
                            </span>
                          </div>
                          <span
                            class="react-resizable-handle"
                          />
                        </th>
                        <th
                          aria-label="Last Name"
                          class="ant-table-cell ant-table-column-has-sorters react-resizable"
                          scope="col"
                          tabindex="0"
                        >
                          <div
                            class="ant-table-filter-column"
                          >
                            <span
                              class="ant-table-column-title"
                            >
                              <div
                                class="ant-table-column-sorters"
                              >
                                <span
                                  class="ant-table-column-title"
                                >
                                  Last Name
                                </span>
                                <span
                                  class="ant-table-column-sorter ant-table-column-sorter-full"
                                >
                                  <span
                                    aria-hidden="true"
                                    class="ant-table-column-sorter-inner"
                                  >
                                    <span
                                      aria-label="caret-up"
                                      class="anticon anticon-caret-up ant-table-column-sorter-up"
                                      role="img"
                                    >
                                      <svg
                                        aria-hidden="true"
                                        data-icon="caret-up"
                                        fill="currentColor"
                                        focusable="false"
                                        height="1em"
                                        viewBox="0 0 1024 1024"
                                        width="1em"
                                      >
                                        <path
                                          d="M858.9 689L530.5 308.2c-9.4-10.9-27.5-10.9-37 0L165.1 689c-12.2 14.2-1.2 35 18.5 35h656.8c19.7 0 30.7-20.8 18.5-35z"
                                        />
                                      </svg>
                                    </span>
                                    <span
                                      aria-label="caret-down"
                                      class="anticon anticon-caret-down ant-table-column-sorter-down"
                                      role="img"
                                    >
                                      <svg
                                        aria-hidden="true"
                                        data-icon="caret-down"
                                        fill="currentColor"
                                        focusable="false"
                                        height="1em"
                                        viewBox="0 0 1024 1024"
                                        width="1em"
                                      >
                                        <path
                                          d="M840.4 300H183.6c-19.7 0-30.7 20.8-18.5 35l328.4 380.8c9.4 10.9 27.5 10.9 37 0L858.9 335c12.2-14.2 1.2-35-18.5-35z"
                                        />
                                      </svg>
                                    </span>
                                  </span>
                                </span>
                              </div>
                            </span>
                            <span
                              class="ant-dropdown-trigger ant-table-filter-trigger"
                              role="button"
                              tabindex="-1"
                            >
                              <span
                                aria-label="filter"
                                class="anticon anticon-filter"
                                role="img"
                              >
                                <svg
                                  aria-hidden="true"
                                  data-icon="filter"
                                  fill="currentColor"
                                  focusable="false"
                                  height="1em"
                                  viewBox="64 64 896 896"
                                  width="1em"
                                >
                                  <path
                                    d="M349 838c0 17.7 14.2 32 31.8 32h262.4c17.6 0 31.8-14.3 31.8-32V642H349v196zm531.1-684H143.9c-24.5 0-39.8 26.7-27.5 48l221.3 376h348.8l221.3-376c12.1-21.3-3.2-48-27.7-48z"
                                  />
                                </svg>
                              </span>
                            </span>
                          </div>
                          <span
                            class="react-resizable-handle"
                          />
                        </th>
                        <th
                          class="ant-table-cell react-resizable"
                          scope="col"
                        >
                          <div
                            class="ant-table-filter-column"
                          >
                            <span
                              class="ant-table-column-title"
                            >
                              Email
                            </span>
                            <span
                              class="ant-dropdown-trigger ant-table-filter-trigger"
                              role="button"
                              tabindex="-1"
                            >
                              <span
                                aria-label="filter"
                                class="anticon anticon-filter"
                                role="img"
                              >
                                <svg
                                  aria-hidden="true"
                                  data-icon="filter"
                                  fill="currentColor"
                                  focusable="false"
                                  height="1em"
                                  viewBox="64 64 896 896"
                                  width="1em"
                                >
                                  <path
                                    d="M349 838c0 17.7 14.2 32 31.8 32h262.4c17.6 0 31.8-14.3 31.8-32V642H349v196zm531.1-684H143.9c-24.5 0-39.8 26.7-27.5 48l221.3 376h348.8l221.3-376c12.1-21.3-3.2-48-27.7-48z"
                                  />
                                </svg>
                              </span>
                            </span>
                          </div>
                          <span
                            class="react-resizable-handle"
                          />
                        </th>
                        <th
                          class="ant-table-cell react-resizable"
                          scope="col"
                        >
                          Password
                          <span
                            class="react-resizable-handle"
                          />
                        </th>
                        <th
                          aria-label="Active"
                          class="ant-table-cell ant-table-column-has-sorters react-resizable"
                          scope="col"
                          tabindex="0"
                        >
                          <div
                            class="ant-table-column-sorters"
                          >
                            <span
                              class="ant-table-column-title"
                            >
                              Active
                            </span>
                            <span
                              class="ant-table-column-sorter ant-table-column-sorter-full"
                            >
                              <span
                                aria-hidden="true"
                                class="ant-table-column-sorter-inner"
                              >
                                <span
                                  aria-label="caret-up"
                                  class="anticon anticon-caret-up ant-table-column-sorter-up"
                                  role="img"
                                >
                                  <svg
                                    aria-hidden="true"
                                    data-icon="caret-up"
                                    fill="currentColor"
                                    focusable="false"
                                    height="1em"
                                    viewBox="0 0 1024 1024"
                                    width="1em"
                                  >
                                    <path
                                      d="M858.9 689L530.5 308.2c-9.4-10.9-27.5-10.9-37 0L165.1 689c-12.2 14.2-1.2 35 18.5 35h656.8c19.7 0 30.7-20.8 18.5-35z"
                                    />
                                  </svg>
                                </span>
                                <span
                                  aria-label="caret-down"
                                  class="anticon anticon-caret-down ant-table-column-sorter-down"
                                  role="img"
                                >
                                  <svg
                                    aria-hidden="true"
                                    data-icon="caret-down"
                                    fill="currentColor"
                                    focusable="false"
                                    height="1em"
                                    viewBox="0 0 1024 1024"
                                    width="1em"
                                  >
                                    <path
                                      d="M840.4 300H183.6c-19.7 0-30.7 20.8-18.5 35l328.4 380.8c9.4 10.9 27.5 10.9 37 0L858.9 335c12.2-14.2 1.2-35-18.5-35z"
                                    />
                                  </svg>
                                </span>
                              </span>
                            </span>
                          </div>
                          <span
                            class="react-resizable-handle"
                          />
                        </th>
                        <th
                          aria-label="Created UTC"
                          class="ant-table-cell ant-table-column-has-sorters react-resizable"
                          scope="col"
                          tabindex="0"
                        >
                          <div
                            class="ant-table-column-sorters"
                          >
                            <span
                              class="ant-table-column-title"
                            >
                              Created UTC
                            </span>
                            <span
                              class="ant-table-column-sorter ant-table-column-sorter-full"
                            >
                              <span
                                aria-hidden="true"
                                class="ant-table-column-sorter-inner"
                              >
                                <span
                                  aria-label="caret-up"
                                  class="anticon anticon-caret-up ant-table-column-sorter-up"
                                  role="img"
                                >
                                  <svg
                                    aria-hidden="true"
                                    data-icon="caret-up"
                                    fill="currentColor"
                                    focusable="false"
                                    height="1em"
                                    viewBox="0 0 1024 1024"
                                    width="1em"
                                  >
                                    <path
                                      d="M858.9 689L530.5 308.2c-9.4-10.9-27.5-10.9-37 0L165.1 689c-12.2 14.2-1.2 35 18.5 35h656.8c19.7 0 30.7-20.8 18.5-35z"
                                    />
                                  </svg>
                                </span>
                                <span
                                  aria-label="caret-down"
                                  class="anticon anticon-caret-down ant-table-column-sorter-down"
                                  role="img"
                                >
                                  <svg
                                    aria-hidden="true"
                                    data-icon="caret-down"
                                    fill="currentColor"
                                    focusable="false"
                                    height="1em"
                                    viewBox="0 0 1024 1024"
                                    width="1em"
                                  >
                                    <path
                                      d="M840.4 300H183.6c-19.7 0-30.7 20.8-18.5 35l328.4 380.8c9.4 10.9 27.5 10.9 37 0L858.9 335c12.2-14.2 1.2-35-18.5-35z"
                                    />
                                  </svg>
                                </span>
                              </span>
                            </span>
                          </div>
                          <span
                            class="react-resizable-handle"
                          />
                        </th>
                        <th
                          class="ant-table-cell react-resizable"
                          scope="col"
                        >
                          Actions
                          <span
                            class="react-resizable-handle"
                          />
                        </th>
                      </tr>
                    </thead>
                    <tbody
                      class="ant-table-tbody"
                    >
                      <tr
                        class="ant-table-row ant-table-row-level-0"
                        data-row-key="00000000-0000-0000-0000-000000000001"
                      >
                        <td
                          class="ant-table-cell"
                        >
                          <div>
                            00000000-0000-0000-0000-000000000001
                          </div>
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          <div>
                            Test
                          </div>
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          <div>
                            User
                          </div>
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          <div>
                            test@example.com
                          </div>
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          <div>
                            password
                          </div>
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          <span
                            aria-label="check-circle"
                            class="anticon anticon-check-circle"
                            role="img"
                            style="color: green;"
                          >
                            <svg
                              aria-hidden="true"
                              data-icon="check-circle"
                              fill="currentColor"
                              focusable="false"
                              height="1em"
                              viewBox="64 64 896 896"
                              width="1em"
                            >
                              <path
                                d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm193.5 301.7l-210.6 292a31.8 31.8 0 01-51.7 0L318.5 484.9c-3.8-5.3 0-12.7 6.5-12.7h46.9c10.2 0 19.9 4.9 25.9 13.3l71.2 98.8 157.2-218c6-8.3 15.6-13.3 25.9-13.3H699c6.5 0 10.3 7.4 6.5 12.7z"
                              />
                            </svg>
                          </span>
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          <div>
                            1st Jan 2024, 05:30
                          </div>
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          <button
                            class="ant-btn css-dev-only-do-not-override-98ntnt ant-btn-text ant-btn-icon-only ant-dropdown-trigger"
                            role="user-action-menu"
                            style="font-size: 16px;"
                            type="button"
                          >
                            <span
                              class="ant-btn-icon"
                            >
                              <span
                                aria-label="more"
                                class="anticon anticon-more"
                                role="img"
                                style="font-size: 20px;"
                              >
                                <svg
                                  aria-hidden="true"
                                  data-icon="more"
                                  fill="currentColor"
                                  focusable="false"
                                  height="1em"
                                  viewBox="64 64 896 896"
                                  width="1em"
                                >
                                  <path
                                    d="M456 231a56 56 0 10112 0 56 56 0 10-112 0zm0 280a56 56 0 10112 0 56 56 0 10-112 0zm0 280a56 56 0 10112 0 56 56 0 10-112 0z"
                                  />
                                </svg>
                              </span>
                            </span>
                          </button>
                        </td>
                      </tr>
                      <tr
                        class="ant-table-row ant-table-row-level-0"
                        data-row-key="00000000-0000-0000-0000-000000000002"
                      >
                        <td
                          class="ant-table-cell"
                        >
                          <div>
                            00000000-0000-0000-0000-000000000002
                          </div>
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          <div>
                            Test
                          </div>
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          <div>
                            User 2
                          </div>
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          <div>
                            test2@example.com
                          </div>
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          <div>
                            password
                          </div>
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          <span
                            aria-label="check-circle"
                            class="anticon anticon-check-circle"
                            role="img"
                            style="color: green;"
                          >
                            <svg
                              aria-hidden="true"
                              data-icon="check-circle"
                              fill="currentColor"
                              focusable="false"
                              height="1em"
                              viewBox="64 64 896 896"
                              width="1em"
                            >
                              <path
                                d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm193.5 301.7l-210.6 292a31.8 31.8 0 01-51.7 0L318.5 484.9c-3.8-5.3 0-12.7 6.5-12.7h46.9c10.2 0 19.9 4.9 25.9 13.3l71.2 98.8 157.2-218c6-8.3 15.6-13.3 25.9-13.3H699c6.5 0 10.3 7.4 6.5 12.7z"
                              />
                            </svg>
                          </span>
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          <div>
                            2nd Jan 2024, 05:30
                          </div>
                        </td>
                        <td
                          class="ant-table-cell"
                        >
                          <button
                            class="ant-btn css-dev-only-do-not-override-98ntnt ant-btn-text ant-btn-icon-only ant-dropdown-trigger"
                            role="user-action-menu"
                            style="font-size: 16px;"
                            type="button"
                          >
                            <span
                              class="ant-btn-icon"
                            >
                              <span
                                aria-label="more"
                                class="anticon anticon-more"
                                role="img"
                                style="font-size: 20px;"
                              >
                                <svg
                                  aria-hidden="true"
                                  data-icon="more"
                                  fill="currentColor"
                                  focusable="false"
                                  height="1em"
                                  viewBox="64 64 896 896"
                                  width="1em"
                                >
                                  <path
                                    d="M456 231a56 56 0 10112 0 56 56 0 10-112 0zm0 280a56 56 0 10112 0 56 56 0 10-112 0zm0 280a56 56 0 10112 0 56 56 0 10-112 0z"
                                  />
                                </svg>
                              </span>
                            </span>
                          </button>
                        </td>
                      </tr>
                    </tbody>
                  </table>
                </div>
              </div>
            </div>
            <ul
              class="ant-pagination ant-table-pagination ant-table-pagination-center css-dev-only-do-not-override-98ntnt"
            >
              <li
                class="ant-pagination-total-text"
              >
                1-2 of 2
              </li>
              <li
                aria-disabled="true"
                class="ant-pagination-prev ant-pagination-disabled"
                title="Previous Page"
              >
                <button
                  class="ant-pagination-item-link"
                  disabled=""
                  tabindex="-1"
                  type="button"
                >
                  <span
                    aria-label="left"
                    class="anticon anticon-left"
                    role="img"
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="left"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M724 218.3V141c0-6.7-7.7-10.4-12.9-6.3L260.3 486.8a31.86 31.86 0 000 50.3l450.8 352.1c5.3 4.1 12.9.4 12.9-6.3v-77.3c0-4.9-2.3-9.6-6.1-12.6l-360-281 360-281.1c3.8-3 6.1-7.7 6.1-12.6z"
                      />
                    </svg>
                  </span>
                </button>
              </li>
              <li
                class="ant-pagination-item ant-pagination-item-1 ant-pagination-item-active"
                tabindex="0"
                title="1"
              >
                <a
                  rel="nofollow"
                >
                  1
                </a>
              </li>
              <li
                aria-disabled="true"
                class="ant-pagination-next ant-pagination-disabled"
                title="Next Page"
              >
                <button
                  class="ant-pagination-item-link"
                  disabled=""
                  tabindex="-1"
                  type="button"
                >
                  <span
                    aria-label="right"
                    class="anticon anticon-right"
                    role="img"
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="right"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z"
                      />
                    </svg>
                  </span>
                </button>
              </li>
              <li
                class="ant-pagination-options"
              >
                <div
                  aria-label="Page Size"
                  class="ant-select ant-select-outlined ant-pagination-options-size-changer css-dev-only-do-not-override-98ntnt ant-select-single ant-select-show-arrow ant-select-show-search"
                >
                  <div
                    class="ant-select-selector"
                  >
                    <span
                      class="ant-select-selection-search"
                    >
                      <input
                        aria-autocomplete="list"
                        aria-controls="rc_select_TEST_OR_SSR_list"
                        aria-expanded="false"
                        aria-haspopup="listbox"
                        aria-label="Page Size"
                        aria-owns="rc_select_TEST_OR_SSR_list"
                        autocomplete="off"
                        class="ant-select-selection-search-input"
                        id="rc_select_TEST_OR_SSR"
                        role="combobox"
                        type="search"
                        value=""
                      />
                    </span>
                    <span
                      class="ant-select-selection-item"
                      title="10 / page"
                    >
                      10 / page
                    </span>
                  </div>
                  <span
                    aria-hidden="true"
                    class="ant-select-arrow"
                    style="user-select: none;"
                    unselectable="on"
                  >
                    <span
                      aria-label="down"
                      class="anticon anticon-down ant-select-suffix"
                      role="img"
                    >
                      <svg
                        aria-hidden="true"
                        data-icon="down"
                        fill="currentColor"
                        focusable="false"
                        height="1em"
                        viewBox="64 64 896 896"
                        width="1em"
                      >
                        <path
                          d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                        />
                      </svg>
                    </span>
                  </span>
                </div>
              </li>
            </ul>
          </div>
        </div>
      </div>
    </div>
  </main>
</div>
`;
